src/main/java/school/hei/haapi/service/WorkDocumentService.java

```diff
@@ -43,32 +41,32 @@ public WorkDocument uploadStudentWorkFile(
       Instant creationDatetime,
       Instant commitmentBegin,
       Instant commitmentEnd,
-      WorkStudyStatus studentWorkStatus,
       MultipartFile workFile) {
     User student = userService.findById(studentId);
 
-    student.setWorkStatus(
-        defineStudentWorkStatus(commitmentBegin, commitmentEnd, studentWorkStatus));
-    student.setCommitmentBeginDate(commitmentBegin);
-    userRepository.save(student);
-
     return fileInfoService.uploadFile(
         student, filename, creationDatetime, commitmentBegin, commitmentEnd, workFile);
   }
 
-  private WorkStudyStatus defineStudentWorkStatus(
-      Instant commitmentBegin, Instant commitmentEnd, WorkStudyStatus studentWorkStatus) {
+  public WorkDocument findLastWorkDocumentByStudentId(String studentId) {
+    return workDocumentRepository
+        .findFirstByStudentIdOrderByCreationDatetimeDesc(studentId)
+        .orElse(new WorkDocument());
```

Comment:
nope, you return Optional also here, and if there's none then you set a different status